Lib/dataclasses.py

```diff
@@ -1122,6 +1123,20 @@ def _dataclass_setstate(self, state):
         object.__setattr__(self, field.name, value)
 
 
+def _get_slots(cls):
+    match cls.__dict__.get('__slots__'):
+        case None:
+            return
+        case str(slot):
+            yield slot
+        # Slots may be any iterable, but we cannot handle an iterator
+        # because it will already be (partially) consumed.
+        case iterable if not hasattr(iterable, '__next__'):
```

Comment:
Is it best to use `'__iter__'` or `'__next__'` here?

Reply:
This trips me up sometimes as well, but `__next__` is the only correct answer, because iter*able*s also have `__iter__`, but only iter*ator*s have `__next__`